catalog/harvest/catalog-harvest-common/src/main/java/org/codice/ddf/catalog/harvest/common/FileSystemPersistenceProvider.java

```diff
@@ -93,16 +96,22 @@ String getMapStorePath() {
   @Override
   public void store(String key, Object value) {
     File dir = new File(getMapStorePath());
-    if (!dir.exists() && !dir.mkdir()) {
-      LOGGER.debug("Unable to create directory: {}", dir.getAbsolutePath());
-    }
-    LOGGER.trace("Entering: store - key: {}", key);
-    try (OutputStream file = new FileOutputStream(getMapStorePath() + key + PERSISTED_FILE_SUFFIX);
-        OutputStream buffer = new BufferedOutputStream(file);
-        ObjectOutputStream output = new ObjectOutputStream(buffer)) {
-      output.writeObject(value);
-    } catch (IOException e) {
-      LOGGER.debug("IOException storing value in cache with key = " + key, e);
+
+    if (new File(getPersistencePath()).canWrite()) {
+      boolean created = dir.mkdirs();
```

Comment:
❓Any reason we can't create this directory once in the constructor instead of making this call every time we write a file to disk?

Reply:
I copied this class over from the CDM module - I'm not sure why it was originally like this, but I would guess this would cover the case where a user deletes the directory and then a caller tries to call store.

Reply:
If the directories already exist, this is equivalent to `dir.exists()`. If any nodes do not exist, this creates them. We could pull up the `dir` itself though. https://docs.oracle.com/javase/7/docs/api/java/io/File.html#mkdirs()

Reply:
This needs to be audited.